opendevin/runtime/runtime.py

```diff
@@ -113,8 +113,7 @@ async def on_event(self, event: Event) -> None:
         if isinstance(event, Action):
             observation = await self.run_action(event)
             observation._cause = event.id  # type: ignore[attr-defined]
-            source = event.source if event.source else EventSource.AGENT
-            await self.event_stream.add_event(observation, source)
+            await self.event_stream.add_event(observation, event.source)  # type: ignore[arg-type]
```

Comment:
Will `event.source` be none here? Shall we enforce some sort of check to make sure every `event.source` has some meaningful value?

Reply:
https://github.com/OpenDevin/OpenDevin/pull/1776#discussion_r1648943935

Reply:
It won't be None here, in the current codebase. Manoj also undid the commit that was making it possible.

That said, I slightly prefer some check too. ("This Should Never Happen(tm) but just in case" 😅)

Reply:
@neubig, waiting for @xingyaoww response.